components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -174,8 +173,13 @@ class DownloadsFeature(
                 startDownload(download)
             }
         } else {
-            onNeedToRequestPermissions(downloadManager.permissions)
-            false
+            if (shouldShowAppDownloaderDialog) {
+                showAppDownloaderDialog(tab, download, apps)
+                return false
+            } else {
+                onNeedToRequestPermissions(downloadManager.permissions)
+                false
+            }
         }
```

Comment:
What about simplifying this block? We only need to check for permissions when third party download ass is not activated, what do you think about the below block?


  ```kotlin
  internal fun processDownload(tab: SessionState, download: DownloadState): Boolean {
        val apps = getDownloaderApps(applicationContext, download)
        // We only show the dialog If we have multiple apps that can handle the download.
        val shouldShowAppDownloaderDialog = shouldForwardToThirdParties() && apps.size > 1
        
        return if (shouldShowAppDownloaderDialog) {
            showAppDownloaderDialog(tab, download, apps)
            false
        } else {
            if (applicationContext.isPermissionGranted(downloadManager.permissions.asIterable())) {
                if (fragmentManager != null && !download.skipConfirmation) {
                    showDownloadDialog(tab, download)
                    false
                } else {
                    useCases.consumeDownload(tab.id, download.id)
                    startDownload(download)
                }
            } else {
                onNeedToRequestPermissions(downloadManager.permissions)
                false
            }
        }
    }
```

Reply:
I confess, i think like 2 hours how can I simplify this sh*t but unable to lol, thanks 😊

Reply:
XD